modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -198,14 +200,12 @@ const setupPurify = (settings: DomParserSettings, schema: Schema): DOMPurifyI =>
     const { attrName, attrValue } = evt;
 
     evt.keepAttr = !validate || Strings.startsWith(attrName, 'data-') || schema.isValid(tagName, attrName);
-    if (attrName in filteredUrlAttrs && !URI.isDomSafe(attrValue, tagName, settings)) {
+    if (!settings.allow_script_urls && attrName in filteredUrlAttrs && URI.isInvalidUri(settings, attrValue, tagName)) {
```

Comment:
The `isDomSafe` function decodes the value passed which isn't needed here, as since this was parsed and looked up from the DOM it's already been decoded which was adding unnecessary overhead.